### PR TITLE
Abhi dutta solutions

### PR DESCRIPTION
**Pull request by abhid94**

### CODE REVIEW

File: Snake/BusinessLayer/Engine.cs [hunk 1]
```diff
@@ -1,203 +1,258 @@
...
-                return false;
-            }
-            if (mazeArray[x, y] == Elements.snakeHead)
-            {
-                return false;
-            }
-            return true;
-        }
-    }
+﻿using BusinessLayer.FoodFolder;
```
Comment (wjoneil) [inline Snake/BusinessLayer/Engine.cs]: Why has every line in this file changed? Line endings?

File: Snake/BusinessLayer/AI.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        private int X;
+        private int Y;
+        Random randomNumber = new Random();
+        public int XCoordinate
+        {
+            get
+            {
+                return X;
+            }
+            set
```
Comment (mattrumble) [inline Snake/BusinessLayer/AI.cs]: Do we ever want anything outside of this class to be able to change toe X or Y coordinates? If not you can remove the set.

File: Snake/BusinessLayer/AI.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        public bool SpawnAI(int XBorder, int YBorder)
+        {
+            if (XBorder < 0 || YBorder < 0)
+            {
+                return false;
+            }
+            this.X = randomNumber.Next(XBorder - 1);
+            this.Y = randomNumber.Next(YBorder - 1);
+           // currentDirection = Direction.Right;
```
Comment (mattrumble) [inline Snake/BusinessLayer/AI.cs]: Remove commented code

File: Snake/BusinessLayer/AI.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+
+        public void MoveAI()
+        {
+
+            bool Opposite = true;
+            int rnd = 0;
+            while (Opposite != false)
+            {
+               // Console.Write("I AM HERE");
```
Comment (mattrumble) [inline Snake/BusinessLayer/AI.cs]: Remove commented code

File: Snake/BusinessLayer/AI.cs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        }
+
+
+
+        public void MoveAI()
+        {
+
+            bool Opposite = true;
+            int rnd = 0;
+            while (Opposite != false)
```
Comment (mattrumble) [inline Snake/BusinessLayer/AI.cs]: while (Opposite)

File: Snake/BusinessLayer/AI.cs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            {
+               // Console.Write("I AM HERE");
+                rnd = randomNumber.Next(0,4);
+                if (rnd == 0 && currentDirection == Direction.Left)
```
Comment (mattrumble) [inline Snake/BusinessLayer/AI.cs]: All these if statements could be a single if statement with || separating each one as they all do the same thing.

File: Snake/BusinessLayer/AI.cs [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+                {
+                    Opposite = false;
+                }
+
+            }
+           
+
+            switch (rnd)
+            {
+                case (int)Direction.Right:
```
Comment (mattrumble) [inline Snake/BusinessLayer/AI.cs]: The casting here feels bad....... we could parse the rnd to the Enum value higher up, that would actually make the if statements in the loop easier to read as they would look like:

if( randomDirection == Direction.Left && currentDirection == Direction.Right)

File: Snake/BusinessLayer/Engine.cs (hunk 1 above)
Comment (abhid94) [inline Snake/BusinessLayer/Engine.cs]: Yes looks like it :(

File: Snake/BusinessLayer/AI.cs (hunk 2 above)
Comment (abhid94) [inline Snake/BusinessLayer/AI.cs]: It turns out this setter is used once in the Engine class

 newAI.XCoordinate = previousX;
 newAI.YCoordinate = previousY;

File: Snake/BusinessLayer/Engine.cs [hunk 8]
```diff
@@ -1,203 +1,258 @@
...
+
+                    gameMaze = new Maze(mazeWidth, mazeLength);
+                    mazeArray = gameMaze.CreateMaze();
+                    gameSound = new GameSound();
+                    gameSnake1 = new GameSnake();
+
+                    List<Point> snakeCurrentBody = gameSnake1.createFirstSnake(mazeLength, mazeWidth, snakeInitialLength);
+                    AddSnakeToTheMaze(snakeCurrentBody);
+
+                   // mazeArray[newAI.XCoordinate, newAI.YCoordinate] = Elements.blank;
```
Comment (mattrumble) [inline Snake/BusinessLayer/Engine.cs]: Remove commented code

File: Snake/BusinessLayer/Engine.cs [hunk 9]
```diff
@@ -1,203 +1,258 @@
...
+
+
+                    mazeArray[newAI.XCoordinate, newAI.YCoordinate] = 0;
+
+                    bool isAIValid = true;
+                    int previousX = newAI.XCoordinate;
+                    int previousY = newAI.YCoordinate;
+                    do
+                    {
+                        newAI.XCoordinate = previousX;
```
Comment (mattrumble) [inline Snake/BusinessLayer/Engine.cs]: These should be done in the MoveAI method I would think

File: Snake/BusinessLayer/AI.cs (hunk 2 above)
Comment (abhid94) [inline Snake/BusinessLayer/AI.cs]: Removed :)

**Pull request merged**